cdragontoolbox/hashes.py

```diff
@@ -722,15 +722,16 @@ def guess_characters_files(self, chars=None):
     def guess_shader_variants(self):
         """Guess different extension variants for shader files, e.g. ".glsl_100" """
 
-        shader_paths = [path for path in self.known.values() if path.endswith(".ps_2_0") or path.endswith(".vs_2_0")]
-
+        shader_paths = [path for path in self.known.values() if os.path.splitext(path)[-1] in [f".{variant}s_{n}_0" for variant in "pv" for n in "23"]]
```

Comment:
You should use a temporary for `[f".{variant}s_{n}_0" for variant in "pv" for n in "23"]`, for readability and to avoid recomputing the list for each path.

Reply:
Good now?